pkcs11/src/new/functions/session_management.rs

```diff
@@ -1,23 +1,51 @@
-use crate::new::types::function::{Result, Rv};
+use crate::new::types::function::Rv;
 use crate::new::types::session::{Session, UserType};
 use crate::new::types::slot_token::Slot;
 use crate::new::types::Flags;
 use crate::new::Pkcs11;
+use crate::new::Result;
+use std::convert::TryInto;
+use std::ffi::CString;
 
 impl Pkcs11 {
-    pub fn open_session(&self, _slot_id: &Slot, _flags: Flags) -> Result<Session> {
-        Err(Rv::Ok)
+    pub fn open_session(&self, slot_id: &Slot, flags: Flags) -> Result<Session> {
+        let mut session_handle = 0;
+
+        Rv::from(unsafe {
+            ((*self.function_list).C_OpenSession.unwrap())(
+                slot_id.id(),
+                flags.into(),
+                // TODO: abstract those types or create new functions for callbacks
```

Comment:
Maybe we should rename this method, `open_session_no_callback` so that the interface for `open_session` doesn't have to change in a breaking way when implementing the callback

Reply:
I think you actually suggested something like that already?

Reply:
ah yes I agree with this way of doing things. I was wondering the reverse, that we would add a method `open_session_with_callback` 😋 But your option makes more sense.